src/main/java/gui/EditRulePopup.java

```diff
@@ -0,0 +1,274 @@
+
+		clearButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				clearMetricsListPanel();
+			}
+		});
+
+		deleteButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (nameText.getText().isEmpty()) {
+					JOptionPane.showMessageDialog(null, "Please insert a rule name!");
+				} else {
+					// Add functionality here.
+				}
+			}
+		});
+
+		saveButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				if (nameText.getText().isEmpty()) {
+					JOptionPane.showMessageDialog(null, "Please insert a rule name!");
+				} else {
+					// Add functionality here.
+				}
+			}
+		});
+
+		controlPanel.add(clearButton);
+		controlPanel.add(deleteButton);
+		controlPanel.add(saveButton);
+		return controlPanel;
+
+	}
+
+	/**
+	 * TODO: The currently * hard-coded values of the JBoxes are to be replaced with
+	 * the correct, final values once the ENUMs are created.
+	 * 
+	 * @return Returns the JPanel holding the line which allows users to add new
+	 *         metrics into the metrics list.
+	 */
+	public JPanel createAddMetricPanel() {
+		addNewMetricPanel.removeAll();
+		addNewMetricPanel.setLayout(new GridLayout(1, 6));
+		condition = new JComboBox<>();
+		setConditionVisibility();
+		JLabel ifCondition = new JLabel("IF", SwingConstants.CENTER);
+		JComboBox<String> value = new JComboBox<>();
+		value.addItem("LOC");
+		value.addItem("MET");
+
+		JComboBox<String> comparison = new JComboBox<>();
+		comparison.addItem(">");
+		comparison.addItem("<");
+		comparison.addItem("=");
+		comparison.addItem("!=");
+
+		JTextField threshold = new JTextField("10");
+
+		JButton addMetricButton = new JButton("Add");
+		addMetricButton.addActionListener(new ActionListener() {
+			public void actionPerformed(ActionEvent e) {
+				String metric;
+				try {
+					Integer.parseInt(threshold.getText());
+				} catch (NumberFormatException ex) {
+					JOptionPane.showMessageDialog(null, "Please check if your threshold input is correct!");
+				}
+				if (ruleMetrics.isEmpty()) {
+					metric = "IF " + value.getSelectedItem() + " " + comparison.getSelectedItem() + " "
+							+ threshold.getText();
+				} else {
+					metric = condition.getSelectedItem() + " " + value.getSelectedItem() + " "
+							+ comparison.getSelectedItem() + " " + threshold.getText();
+				}
+				ruleMetrics.add(metric);
+				fillMetricsListPanel();
+				setConditionVisibility();
+				metricsListPanel.revalidate();
+				metricsListPanel.repaint();
+			}
+		});
+
+		addNewMetricPanel.add(condition);
+		addNewMetricPanel.add(ifCondition);
+		addNewMetricPanel.add(value);
+		addNewMetricPanel.add(comparison);
+		addNewMetricPanel.add(threshold);
+		addNewMetricPanel.add(addMetricButton);
+		addNewMetricPanel.setPreferredSize(new Dimension(650, 25));
+		addNewMetricPanel.setMaximumSize(new Dimension(650, 25));
+		return addNewMetricPanel;

```

Comment:
I don't know which part of the code does this, and this is just a nice to have. But it would be nice if the lines of the added rules would be centred in the middle of the panel, instead of being equally spaced.

<img width="777" alt="image" src="https://user-images.githubusercontent.com/14364481/69367189-b2edb180-0c8f-11ea-95ab-6ebb0b030c7f.png">
<img width="778" alt="image" src="https://user-images.githubusercontent.com/14364481/69367280-e3355000-0c8f-11ea-9078-cc44a765a2de.png">

Reply:
@ftfni-iscteiul open this as an increment for improvement - don't shoulder @Hugo-Barroca with this in this task x)

Reply:
ok